api.planx.uk/helpers.ts

```diff
@@ -66,55 +66,96 @@ interface CreateFlowResponse {
   };
 }
 
+const createFlowMutation = gql`
+  mutation InsertFlow(
+    $team_id: Int!
+    $slug: String!
+    $name: String!
+    $data: jsonb = {}
+    $copied_from: uuid
+    $templated_from: uuid
+  ) {
+    flow: insert_flows_one(
+      object: {
+        team_id: $team_id
+        slug: $slug
+        name: $name
+        data: $data
+        version: 1
+        copied_from: $copied_from
+        templated_from: $templated_from
+      }
+    ) {
+      id
+    }
+  }
+`;
+
+const createSourceTemplateMutation = gql`
+  mutation InsertSourceTemplateFlow(
+    $team_id: Int!
+    $slug: String!
+    $name: String!
+    $data: jsonb = {}
+    $copied_from: uuid
+    $templated_from: uuid
+    $is_template: Boolean
+  ) {
+    flow: insert_flows_one(
+      object: {
+        team_id: $team_id
+        slug: $slug
+        name: $name
+        data: $data
+        version: 1
+        copied_from: $copied_from
+        templated_from: $templated_from
+        is_template: $is_template
+      }
+    ) {
+      id
+    }
+  }
+`;
+
 // Insert a new flow into the `flows` table
-const createFlow = async (
-  teamId: number,
-  slug: string,
-  name: string,
-  flowData: Flow["data"],
-  copiedFrom?: Flow["id"],
-  templatedFrom?: Flow["id"],
-) => {
+const createFlow = async ({
+  teamId,
+  slug,
+  name,
+  isTemplate,
+  flowData,
+  copiedFrom,
+  templatedFrom,
+}: {
+  teamId: number;
+  slug: string;
+  name: string;
+  isTemplate: boolean;
+  flowData: Flow["data"];
+  copiedFrom?: Flow["id"];
+  templatedFrom?: Flow["id"];
+}) => {
   const { client: $client } = getClient();
   const userId = userContext.getStore()?.user?.sub;
 
+  /** Only platformAdmins have write permissions for the is_template column */
+  const mutation = isTemplate
+    ? createSourceTemplateMutation
+    : createFlowMutation;
```

Comment:
This feels slightly awkward, not sure if there's a better pattern we could use here?